thrust/thrust/detail/contiguous_storage.h

```diff
@@ -100,7 +111,33 @@ class contiguous_storage
 
   _CCCL_HOST_DEVICE void deallocate() noexcept;
 
-  _CCCL_HOST_DEVICE void swap(contiguous_storage& x);
+  _CCCL_EXEC_CHECK_DISABLE
+  _CCCL_HOST_DEVICE void swap(contiguous_storage& x) noexcept(
+    allocator_traits<Alloc>::propagate_on_container_swap::value || allocator_traits<Alloc>::is_always_equal::value)
```

Comment:
That's probably wrong, because in theory swapping any of the members below could throw.

Reply:
I dropped the `noexcept`.